apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -30,12 +31,19 @@ defmodule Aecore.Structures.SignedTx do
 
   @spec is_valid?(SignedTx.t()) :: boolean()
   def is_valid?(tx) do
-    tx.data.value >= 0 && tx.data.fee >= 0 && Keys.verify_tx(tx)
+
+    case tx.data do
```

Comment:
Continuing with our proposal with @rdimitrovbg here as well. 

Here we can make each Transaction module have it's own is_valid?() func and call it from there like: 

```elixir
defmodule Aecore.Structures.SignedTx do
.
.
.
    def is_valid?(%SignedTx{data: data_tx} = tx) do
        Aecore.Structures.DataTx.is_valid?(data_tx)
    end
end 
```

```elixir
defmodule Aecore.Structures.DataTx do
.
.
.
    def is_valid?(%DataTx{type: type, payload: payload, fee: fee} = data_tx) do
        if (tx.data.fee >= 0 && Keys.verify_tx(tx)) do
            type.is_valid?(payload)
        else
            false
        end
    end
end 
```

```elixir
defmodule Aecore.Structures.SpendTx do
.
.
.
    def is_valid?(%SpendTx{value: value} = payload) do
        tx.data.value >= 0
    end
end 
```